fedimint-testing/src/ln/mock.rs

```diff
@@ -134,70 +130,3 @@ impl ILnRpcClient for FakeLightningTest {
         Ok(Box::pin(stream::iter(vec![])))
     }
 }
-
-/// A proxy for the underlying LnRpc which can be used to add behavior to it
-/// using the "Decorator pattern"
-#[derive(Debug, Clone)]
-pub struct LnRpcAdapter {
-    /// The actual `ILnRpcClient` that we add behavior to.
-    client: Arc<RwLock<dyn ILnRpcClient>>,
-    /// A pair of [`PayInvoiceRequest`] and `u8` where client.pay() will fail
-    /// `u8` times for each `String` (bolt11 invoice)
-    fail_invoices: Arc<sync::Mutex<HashMap<String, u8>>>,
-}
-
-impl LnRpcAdapter {
-    pub fn new(client: Arc<RwLock<dyn ILnRpcClient>>) -> Self {
-        let fail_invoices = Arc::new(sync::Mutex::new(HashMap::new()));
-
-        LnRpcAdapter {
-            client,
-            fail_invoices,
-        }
-    }
-
-    /// Register `invoice` to fail `times` before (attempt) succeeding. The
-    /// invoice will be dropped from the HashMap after succeeding
-    #[allow(dead_code)]
-    pub async fn fail_invoice(&self, invoice: PayInvoiceRequest, times: u8) {
-        self.fail_invoices
-            .lock()
-            .await
-            .insert(invoice.invoice, times + 1);
-    }
```

Comment:
So we had a mechanism for failing invoices but we never used it anywhere? cc @okjodom

Reply:
Yeah I don't think it is used. If we need that functionality I don't think we need another adapter class, we can use add it to `LightningTest` instead